src/plugins/autocompletion_providers/Git.ts

```diff
@@ -84,181 +141,310 @@ const notStagedFiles = unique(async(context: AutocompletionContext): Promise<Sug
     }
 });
 
-const subCommandProviders: Dictionary<AutocompletionProvider> = {
-    add: combine([notStagedFiles, addOptions]),
-    checkout: combine([branchesExceptCurrent, branchAlias, notStagedFiles]),
-    commit: commitOptions,
-    status: statusOptions,
-    merge: combine([branchesExceptCurrent, branchAlias]),
-    push: pushOptions,
-    config: combine([configVariables, configOptions]),
-};
+interface GitCommandData {
+    name: string;
+    description: string;
+    provider?: AutocompletionProvider;
```

Comment:
It would be better to create an `empty` provider and use it everywhere than having an optional property.